src/main/java/ca/corefacility/bioinformatics/irida/ria/web/projects/settings/ProjectSettingsRemoteAjaxController.java

```diff
@@ -0,0 +1,63 @@
+package ca.corefacility.bioinformatics.irida.ria.web.projects.settings;
+
+import java.security.Principal;
+import java.util.Locale;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import ca.corefacility.bioinformatics.irida.model.project.Project;
+import ca.corefacility.bioinformatics.irida.ria.web.ajax.dto.RemoteProjectSettings;
+import ca.corefacility.bioinformatics.irida.ria.web.ajax.dto.RemoteProjectSettingsUpdateRequest;
+import ca.corefacility.bioinformatics.irida.ria.web.ajax.dto.ajax.AjaxResponse;
+import ca.corefacility.bioinformatics.irida.ria.web.services.UIRemoteProjectService;
+
+/**
+ * Controller for managing settings for a remotely sync'd project
+ * asynchronously
+ */
+@RestController
+@RequestMapping("/ajax/remote-projects/{projectId}/settings")
+public class ProjectSettingsRemoteAjaxController {
+	private UIRemoteProjectService uiRemoteProjectService;
+
+	@Autowired
+	public ProjectSettingsRemoteAjaxController(UIRemoteProjectService uiRemoteProjectService) {
+		this.uiRemoteProjectService = uiRemoteProjectService;
+	}
+
+	/**
+	 * Update the remote project sync settings
+	 *
+	 * @param projectId                          the project id to update
+	 * @param remoteProjectSettingsUpdateRequest object which
+	 *                                           is used to update frequency, and sync user, as well as
+	 *                                           force sync for a project
+	 * @param principal                          The current logged in user
+	 * @param locale                             user's locale
+	 * @return AjaxResponse if error return AjaxErrorResponse otherwise return AjaxUpdateItemSuccessResponse
+	 */
+	@PostMapping(value = "/sync")
+	public ResponseEntity<AjaxResponse> updateProjectSyncSettings(@PathVariable Long projectId,
+			@RequestBody RemoteProjectSettingsUpdateRequest remoteProjectSettingsUpdateRequest, Principal principal,
```

Comment:
You should handle this a little differently.  Have your service throw an error catch it here and then create your responses.  The reason is that if it fails you can return an `HttpStatus`. error code.

Reply:
Updated in dc6a7340bb060bcf798d856d3e29b7da3fff9389